liverpooltickets/download_data.py

```
@@ -0,0 +1,44 @@
+from typing import Set
+from urllib.request import urlopen, urlretrieve
+import re
+import os
+
+
+def get_csv_urls_from_page() -> Set[str]:
+    """Finds and returns all the urls to download the csv files for the Premier League games
+
+    :returns: a set of urls from where the csv files can be downloaded
+    :rtype: set
+    """
+    with urlopen('https://www.football-data.co.uk/englandm.php') as response:
+        premier_league_regex = '<A HREF=".+">Premier League</A>'
+        lines = {line.decode('utf-8') for line in response}
+        urls = '\n'.join(lines)
+        urls = re.findall(premier_league_regex, urls)
+        urls = {'https://www.football-data.co.uk/' + re.split('"', url)[1] for url in urls}
+        return urls
+
+
+def download_csv_files(urls: Set[str], folder_path: str = '../data') -> None:
+    """Download and save the csv files from a list of urls in a folder, that is created if it doesn't yet exist
+
+    :param urls: The set of urls to download the csv files from
+    :type urls: set
+    :param folder_path: The path to the folder where the files should be saved
+    :type folder_path: str
+    """
+    if not os.path.exists(folder_path):  # Create a directory if it does not exist.
```

there is an option of the os.makedirs function to fail silently in case it already exists ;)